src/System.Net.Primitives/src/System/Net/CookieContainer.cs

```diff
@@ -0,0 +1,1062 @@
+            {
+                return true;
+            }
+
+            // Test for "127.###.###.###" without using regex.
+            string[] ipParts = host.Split('.');
+            if (ipParts != null && ipParts.Length == 4 && ipParts[0] == "127")
+            {
+                int i;
+                for (i = 1; i < 4; i++)
+                {
+                    switch (ipParts[i].Length)
+                    {
+                        case 3:
+                            if (ipParts[i][2] < '0' || ipParts[i][2] > '9')
+                            {
+                                break;
+                            }
+                            goto case 2;
+
+                        case 2:
+                            if (ipParts[i][1] < '0' || ipParts[i][1] > '9')
+                            {
+                                break;
+                            }
+                            goto case 1;
+
+                        case 1:
+                            if (ipParts[i][0] < '0' || ipParts[i][0] > '9')
+                            {
+                                break;
+                            }
+                            continue;
+                    }
+                    break;
+                }
+                if (i == 4)
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        public void Add(Uri uri, Cookie cookie)
+        {
+            if (uri == null)
+            {
+                throw new ArgumentNullException("uri");
+            }
+            if (cookie == null)
+            {
+                throw new ArgumentNullException("cookie");
+            }
+            Cookie new_cookie = cookie.Clone();
+            new_cookie.VerifySetDefaults(new_cookie.Variant, uri, IsLocalDomain(uri.Host), _fqdnMyDomain, true, true);
+
+            Add(new_cookie, true);
+        }
+
+        public void Add(Uri uri, CookieCollection cookies)
+        {
+            if (uri == null)
+            {
+                throw new ArgumentNullException("uri");
+            }
+            if (cookies == null)
+            {
+                throw new ArgumentNullException("cookies");
+            }
+
+            bool isLocalDomain = IsLocalDomain(uri.Host);
+            foreach (Cookie c in cookies)
+            {
+                Cookie new_cookie = c.Clone();
+                new_cookie.VerifySetDefaults(new_cookie.Variant, uri, isLocalDomain, _fqdnMyDomain, true, true);
+                Add(new_cookie, true);
+            }
+        }
+
+        internal CookieCollection CookieCutter(Uri uri, string headerName, string setCookieHeader, bool isThrow)
+        {
+            GlobalLog.Print("CookieContainer#" + Logging.HashString(this) + "::CookieCutter() uri:" + uri + " headerName:" + headerName + " setCookieHeader:" + setCookieHeader + " isThrow:" + isThrow);
```

Comment:
/cc @josguil

All above performance improvements for logging mentioned by @stephentoub will be tracked by #2500.